src/PHPStan/Reflection/TypeToCallReflectionResolver/ClosureTypeToCallReflectionResolver.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\PHPStan\Reflection\TypeToCallReflectionResolver;
+
+use PHPStan\Reflection\ClassMemberAccessAnswerer;
+use PHPStan\Reflection\Native\NativeFunctionReflection;
+use PHPStan\TrinaryLogic;
+use PHPStan\Type\ClosureType;
+use PHPStan\Type\Type;
+
+final class ClosureTypeToCallReflectionResolver implements TypeToCallReflectionResolverInterface
+{
+    public static function getType(): string
+    {
+        return ClosureType::class;
+    }
+
+    public function resolve(Type $type, ClassMemberAccessAnswerer $classMemberAccessAnswerer)
+    {
+        if (! $type instanceof ClosureType) {
+            return null;
+        }
+
```

Comment:
If you add `isMatch()` method, it would separate decide logic and perform logic.
It would make code more clear about contracts and less loosen